File: js-python-educative-challenges/two-pointers/palindrome_string.py
def is_palindrome(s: str) -> bool:
    """Takes a string and determines whether or not it is a palindrome."""
    if len(s) < 1:
        return False
    for i in range(len(s)):
        if i < abs(-1-i):
            one = s[i]
            two = s[-1-i]
            if one != two:
                return False
    return True

if __name__ == '__main__':
    s = 'moomk'
    print(is_palindrome(s))

File: js-python-educative-challenges/two-pointers/palindrome_string.js
/**
 * Takes a string and determines whether or not it is a palindrome.
 * @param {string} s
 * @returns {boolean}
 */

function isPalindrome(s) {
  if (s.length < 1) return false;
  for (let i = 0; i < s.length; i++) {
    if (i < Math.abs(-1-i)) {
      const one = s[i]
      const two = s.at(-1-i)
      if (one !== two) return false;
    }
  }
  return true
}

// main

const s = "moomk"
console.log(isPalindrome(s))
